pkg/controllers/gateway/suite_test.go

```diff
@@ -262,234 +275,349 @@ var _ = Describe("GatewayClassController", func() {
-// 		})
-// 	})
-// })
+var _ = Describe("GatewayController", func() {
+	Context("testing gateway controller", func() {
+		var gateway *gatewayv1beta1.Gateway
+		var noLGateway *gatewayv1beta1.Gateway
+		var gatewayclass *gatewayv1beta1.GatewayClass
+		var managedZone *v1alpha1.ManagedZone
+		var manifest1 *ocmworkv1.ManifestWork
+		var manifest2 *ocmworkv1.ManifestWork
+		var nsSpoke1 *corev1.Namespace
+		var nsSpoke2 *corev1.Namespace
+		var placementDecision *ocmclusterv1beta1.PlacementDecision
+		var tlsSecrets *corev1.Secret
+		var hostnametest gatewayv1beta1.Hostname
+		var stringified string
+
+		//Kicks off before the tests run, used to create resources we need in multiple tests,were created in a diff function that we are not testing but need the output, or things a user may have to do.
+		BeforeEach(func() {
+			// Before: Create GatewayClass for the gateway
+			gatewayclass = &gatewayv1beta1.GatewayClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "kuadrant-multi-cluster-gateway-instance-per-cluster",
+					Namespace: defaultNS,
+				},
+				Spec: gatewayv1beta1.GatewayClassSpec{
+					ControllerName: "kuadrant.io/mctc-gw-controller",
+				},
+			}
+			Expect(k8sClient.Create(ctx, gatewayclass)).To(BeNil())
+
+			// Before: Create a test ManagedZone for test Gateway listeners to use
+			managedZone = &v1alpha1.ManagedZone{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "example.com",
+					Namespace: defaultNS,
+				},
+				Spec: v1alpha1.ManagedZoneSpec{
+					ID:          "1234",
+					DomainName:  "example.com",
+					Description: "example.com",
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, managedZone)).To(BeNil())
+
+			//Before: Create placement decision
+
+			placementDecision = &ocmclusterv1beta1.PlacementDecision{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-placement",
+					Namespace: defaultNS,
+					Labels: map[string]string{
+						"cluster.open-cluster-management.io/placement": "true",
+					},
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, placementDecision)).To(BeNil())
+
+			// Before: Update the statsus of theplacement decision to include the clusters we want to place as OCM would usually do this
+			placementDecision.Status = ocmclusterv1beta1.PlacementDecisionStatus{
+				Decisions: []ocmclusterv1beta1.ClusterDecision{
+					{
+						ClusterName: nsSpoke1Name,
+						Reason:      "test",
+					},
+					{
+						ClusterName: nsSpoke2Name,
+						Reason:      "test",
+					},
+				},
+			}
+			Expect(k8sClient.Status().Update(ctx, placementDecision)).To(BeNil())
+			//Before: Stub Gateway for tests but not creating it here
+			hostname := gatewayv1beta1.Hostname("test.example.com")
+			gateway = &gatewayv1beta1.Gateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-gw-1",
+					Namespace: defaultNS,
+					Labels: map[string]string{
+						"cluster.open-cluster-management.io/placement": "true",
```

Comment:
The test works as is with the value of 'true' for this label,
but I'm wondering if a more meaning value would be better in the test?
Something like 'GatewayControllerTest'? (can be anything)
The value of 'true' might give the impression that it can only be true or false, which is not the case i.e. it's an arbitrary value that must match between the Gateway and the PlacementDecision